cmd/liqonet/route-operator.go

```diff
@@ -0,0 +1,171 @@
+		"VXLAN Virtual Tunnel Endpoints (VTEP) port for the Liqonet intra-cluster overlay network")
+}
+
+func runRouteOperator(commonFlags *liqonetCommonFlags, routeFlags *routeOperatorFlags) {
+	vxlanMTU := routeFlags.Mtu
+	vTepPort := routeFlags.vni
+	VNI := routeFlags.VtepPort
+	metricsAddr := commonFlags.metricsAddr
+	vxlanConfig := &overlay.VxlanDeviceAttrs{
+		Vni:      VNI,
+		Name:     liqoconst.VxlanDeviceName,
+		VtepPort: vTepPort,
+		VtepAddr: nil,
+		MTU:      vxlanMTU,
+	}
+
+	mutex := &sync.RWMutex{}
+	nodeMap := map[string]string{}
+	// Get the pod ip and parse to net.IP.
+	podIP, err := utils.GetPodIP()
+	if err != nil {
+		klog.Errorf("unable to get podIP: %v", err)
+		os.Exit(1)
+	}
+	nodeName, err := utils.GetNodeName()
+	if err != nil {
+		klog.Errorf("unable to get node name: %v", err)
+		os.Exit(1)
+	}
+	podNamespace, err := utils.GetPodNamespace()
+	if err != nil {
+		klog.Errorf("unable to get pod namespace: %v", err)
+		os.Exit(1)
+	}
+	// Asking the api-server to only inform the operator for the pods running in a node different from the one
+	// where the operator is running.
+	smcFieldSelector, err := fields.ParseSelector(strings.Join([]string{"spec.nodeName", "!=", nodeName}, ""))
+	if err != nil {
+		klog.Errorf("unable to create label requirement: %v", err)
+		os.Exit(1)
+	}
+	// Asking the api-server to only inform the operator for the pods running in a node different from
+	// the virtual nodes. We want to process only the pods running on the local cluster and not the ones
+	// offloaded to a remote cluster.
+	smcLabelRequirement, err := labels.NewRequirement(liqoconst.LocalPodLabelKey, selection.DoesNotExist, []string{})
+	if err != nil {
+		klog.Errorf("unable to create label requirement: %v", err)
+		os.Exit(1)
+	}
+	smcLabelSelector := labels.NewSelector().Add(*smcLabelRequirement)
+	mainMgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		MapperProvider:     mapperUtils.LiqoMapperProvider(scheme),
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
+		NewCache: cache.BuilderWithOptions(cache.Options{
+			SelectorsByObject: cache.SelectorsByObject{
+				&corev1.Pod{}: {
+					Field: smcFieldSelector,
+					Label: smcLabelSelector,
+				},
+			},
+		}),
+	})
+	if err != nil {
+		klog.Errorf("unable to get manager: %s", err)
+		os.Exit(1)
+	}
+	// Asking the api-server to only inform the operator for the pods that are part of the route component.
+	ovcLabelSelector := labels.SelectorFromSet(labels.Set{
+		podNameLabelKey:     routeNameLabelValue,
+		podInstanceLabelKey: routeInstanceLabelValue,
+	})
+	// This manager is used by the overlay operator and it is limited to the pods running
+	// on the same namespace as the operator.
+	overlayMgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		MapperProvider:     mapperUtils.LiqoMapperProvider(scheme),
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
```

Comment:
Should be tested, but I had issues in the past with two managers binding to the same metrics port